src/operator/subgraph/build_subgraph.cc

```diff
@@ -749,6 +749,17 @@ void CreateSubgraphNode(nnvm::Graph* g,
         for (BiDirectedNode* dest_node : subgraph_nodes) {
           sn->outputs.erase(dest_node->node);
         }
+      }
+    }
+
+    // Set outputs according to current inputs
+    for (size_t i = 0; i < n->inputs.size(); ++i) {
+      auto& e = n->inputs[i];
+      // update input entries' source simple nodes' outputs map
+      nnvm::Node* node = e.node.get();
+      if (indexed_graph.exist(node)) {
+        const auto nid     = indexed_graph.node_id(node);
+        BiDirectedNode* sn = simple_nodes[nid].get();
```

Comment:
Do these changes here actually do anything? `sn` is not used anywhere.

Reply:
Yes, ouput is add in line 763 using `sn`.
The differense between porevious version is that first all nodes which should be removed  are removed in line 750
and later all node which sould be added are added in line 763.